app/src/test/java/org/simple/clinic/facility/change/FacilityChangeScreenControllerTest.kt

```diff
@@ -273,4 +273,65 @@ class FacilityChangeScreenControllerTest {
   fun `params for permission denials`(): List<RuntimePermissionResult> {
     return listOf(DENIED, NEVER_ASK_AGAIN)
   }
+
+  @Test
+  @Parameters("6", "7", "8")
+  fun `when location is being fetched then it should expire after a fixed time duration`(
+      secondsSpentWaitingForLocation: Long
+  ) {
+    configProvider.onNext(configTemplate.copy(locationListenerExpiry = Duration.ofSeconds(5)))
+
+    whenever(facilityRepository.facilitiesInCurrentGroup(any(), any())).thenReturn(Observable.just(emptyList()))
+    whenever(locationRepository.streamUserLocation(any(), any())).thenReturn(Observable.never())
+
+    uiEvents.onNext(ScreenCreated())
+    uiEvents.onNext(FacilityChangeLocationPermissionChanged(GRANTED))
+    verify(screen).showProgressIndicator()
+
+    testComputationScheduler.advanceTimeBy(secondsSpentWaitingForLocation, SECONDS)
+    verify(screen).hideProgressIndicator()
+  }
+
+  @Test
+  fun `when screen starts and location permission is available then progress indicator should be shown`() {
+    whenever(locationRepository.streamUserLocation(any(), any())).thenReturn(Observable.never())
+
+    uiEvents.onNext(ScreenCreated())
+    uiEvents.onNext(FacilityChangeLocationPermissionChanged(GRANTED))
+    uiEvents.onNext(FacilityChangeLocationPermissionChanged(GRANTED))
```

Comment:
I think there should also be a test for when the permission changes from one state to another i.e. `DENIED` -> `GRANTED`.
What do you think?

Reply:
I actually started with that, but realized that according to the spec, we will not observe changes in permission. Only the first result will be picked so it's not required. I'm emitting multiple `GRANTED` values here just as a sanity check to ensure `showProgressIndicator()` doesn't get called twice.